zoninator.php

```diff
@@ -1332,51 +611,84 @@ function do_zoninator_feeds() {
 
 		if ( ! empty( $query_var ) ) {
-			$zone_slug = get_query_var( $this->zone_taxonomy );
-			$zone_id = $this->get_zone( $zone_slug );
+			$zone_slug = get_query_var( Zoninator_Constants::ZONE_TAXONOMY );
 
-			if ( empty( $zone_id ) ) {
-				$this->send_user_error( __( 'Invalid zone supplied', 'zoninator' ) );
-			}
-
-			$results = $this->get_zone_posts( $zone_id, apply_filters( 'zoninator_json_feed_fields', array(), $zone_slug ) );
+			$results = $this->zone_gateway->get_zone_feed( $zone_slug );
 
-			if ( empty( $results ) ) {
-				$this->send_user_error( __( 'No zone posts found', 'zoninator' ) );
+			if ( is_wp_error( $results ) ) {
+				$this->send_user_error( $results->get_error_message() );
 			}
 
-			$filtered_results = $this->filter_zone_feed_fields( $results );
-
-			$this->json_return( apply_filters( 'zoninator_json_feed_results', $filtered_results, $zone_slug ), false );
+			$this->json_return( $results );
 		}
 
 		return;
 
 	}
 
-	private function filter_zone_feed_fields( $results ) {
+	/**
+	 * Attached to activate_{ plugin_basename( __FILES__ ) } by register_activation_hook()
+	 * @static
+	 */
+	public static function plugin_activation() {
+		if ( version_compare( $GLOBALS['wp_version'], ZONINATOR_MINIMUM_WP_VERSION, '<' ) ) {
+			load_plugin_textdomain( 'zoninator' );
 
-		$whitelisted_fields = array( 'ID', 'post_date', 'post_title', 'post_content', 'post_excerpt', 'post_status', 'guid' );
+			$message = '<strong>'.sprintf(esc_html__( 'Zoninator %s requires WordPress %s or higher.' , 'zoninator'), ZONINATOR_VERSION, ZONINATOR_MINIMUM_WP_VERSION ).'</strong> '.sprintf(__('Please <a href="%1$s">upgrade WordPress</a> to a current version, or <a href="%2$s">downgrade to version 0.6 of the Zoninator plugin</a>.', 'zoninator'), 'https://codex.wordpress.org/Upgrading_WordPress', 'https://wordpress.org/plugins/zoninator/download');
 
-		$i = 0;
-		foreach ( $results as $result ) {
-			foreach( $whitelisted_fields as $field ) {
-					$filtered_results[$i]->$field = $result->$field;
-			}
-			$i++;
+			self::bail_on_activation( $message, true );
 		}
+	}
 
+	private static function bail_on_activation( $message, $deactivate = true ) {
+?>
+<!doctype html>
+<html>
+<head>
+<meta charset="<?php bloginfo( 'charset' ); ?>">
+<style>
+* {
+	text-align: center;
+	margin: 0;
+	padding: 0;
+	font-family: "Lucida Grande",Verdana,Arial,"Bitstream Vera Sans",sans-serif;
+}
+p {
+	margin-top: 1em;
+	font-size: 18px;
+}
+</style>
+<body>
+<p><?php echo esc_html( $message ); ?></p>
+</body>
+</html>
+<?php
+		if ( $deactivate ) {
+			$plugins = get_option( 'active_plugins' );
+			$_zoninator = plugin_basename( ZONINATOR_PLUGIN_DIR . 'zoninator.php' );
+			$update  = false;
+			foreach ( $plugins as $i => $plugin ) {
+				if ( $plugin === $_zoninator ) {
+					$plugins[$i] = false;
+					$update = true;
+				}
+			}
 
-		return $filtered_results;
+			if ( $update ) {
+				update_option( 'active_plugins', array_filter( $plugins ) );
```

Comment:
Can we use https://developer.wordpress.org/reference/functions/deactivate_plugins/ instead?